src/windows-emulator/windows_emulator.cpp

```diff
@@ -515,14 +515,23 @@ void windows_emulator::setup_hooks()
 
     this->emu().hook_interrupt([&](const int interrupt) {
         const auto rip = this->emu().read_instruction_pointer();
+        const auto eflags = this->emu().reg<uint32_t>(x86_register::eflags);
 
         switch (interrupt)
         {
         case 0:
             dispatch_integer_division_by_zero(this->emu(), this->process);
             return;
         case 1:
-            this->log.print(color::pink, "Singlestep: 0x%" PRIx64 "\n", rip);
+            if ((eflags & 0x100) != 0)
+            {
+                this->log.print(color::pink, "Singlestep (Trap Flag): 0x%" PRIx64 "\n", rip);
+                this->emu().reg(x86_register::eflags, eflags & ~0x100);
```

Comment:
are you sure removing that is correct?

Reply:
Hm, that's a good question. Actually it looks like it should only be unset during the exception handler.

Reply:
Okay, I did some more in-depth testing, and it looks like removing the trap bit is the correct thing to do. I added a test for it just in case, and it passes on both the emulator and my host machine.